built_in_tasks/force_task.py

```diff
@@ -0,0 +1,178 @@
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(self, *args, **kwargs)
+        self.disk_cursor = VirtualCircularTarget(target_radius=0, target_color=disk_colors[self.disk_cursor_color], starting_pos=self.disk_pos)
+        self.disk_target = VirtualTorusTarget() # just a placeholder
+        for model in self.disk_cursor.graphics_models:
+            self.add_model(model)
+        self.hold_penalty = False
+        np.random.seed() # initialize random number generator
+
+    def init(self):
+        self.add_dtype('disk_cursor_size', 'f8', (1,))
+        self.add_dtype('disk_target_size', 'f8', (1,))
+        super().init()
+
+    def update_disk_cursor(self):
+        
+        # Remove the previous disk cursor
+        for model in self.disk_cursor.graphics_models:
+            self.remove_model(model)
+
+        # Add a new cursor with updated radius
+        raw_value = self.joystick.get()
+        if raw_value is None:
+            raw_value = 0.09
+        cm_value = (raw_value + self.disk_cursor_offset) * self.disk_cursor_gain
+
+        self.disk_cursor_radius = min(max(cm_value, self.disk_cursor_bounds[0]), self.disk_cursor_bounds[1])
+        self.task_data['disk_cursor_size'] = self.disk_cursor_radius
+        self.disk_cursor = VirtualCircularTarget(target_radius=self.disk_cursor_radius, target_color=disk_colors[self.disk_cursor_color], starting_pos=self.disk_pos)
+        for model in self.disk_cursor.graphics_models:
+            self.add_model(model)
+
+        if self.state == "pause":
+            self.disk_cursor.hide()
+   
+    #### TEST FUNCTIONS ####
+    def _test_start_trial(self, ts):
+        return not self.pause
+
+    def _test_enter_target(self, ts):
+        d = abs(self.disk_cursor_radius - self.disk_target_radius_trial)
+        return d <= self.disk_target_tolerance
+
+    def _test_leave_target(self, ts):
+        d = abs(self.disk_cursor_radius - self.disk_target_radius_trial)
+        return d > self.disk_target_tolerance
+
+    def _test_hold_complete(self, ts):
+        return ts > self.hold_time_trial
+    
+    def _test_hold_penalty_end(self, ts):
+        return ts > self.hold_penalty_time
+
+    def _test_reward_end(self, ts):
+        return ts > self.reward_time
+    
+    def _test_start_pause(self, ts):
+        return self.pause
+    
+    def _test_continue_pause(self, ts):
+        return self.pause
+    
+    def _test_end_pause(self, ts):
+        return not self.pause
+
+    ### STATE FUNCTIONS ###
+    def _start_wait(self):
+
+        # Select a random parameters
+        if not self.hold_penalty:
+            self.disk_target_radius_trial = np.random.uniform(*self.disk_target_radius)
+            self.hold_time_trial = np.random.uniform(*self.hold_time)
+            print("New target radius:", self.disk_target_radius_trial)
+
+    def _start_target(self):
+        self.sync_event('TARGET_ON', 0)
+        
```

Comment:
Can this event code be used even after you combine this task with the center out task?